README.md

```diff
@@ -20,52 +20,80 @@ For requestor the app should implement a long running RPC service which implemen
 
 ### RPC methods
 - `CreateTask`
-  - takes two arguments `task_id` and `task_params_json`
-  - should treat `{work_dir}/{task_id}` as the working directory for the given task
-  - `task_params_json` is a JSON string containing task specific parameters
-  - will only be called once with given `task_id`
-  - can assume `{task_id}/{constants.TASK_INPUTS_DIR}` contains all the resources provided by task creator
+  - Takes three arguments: `task_id`, `max_subtasks_count`, and `task_params_json`.
+  - Should treat `{work_dir}/{task_id}` as the working directory for the given task.
+  - `task_params_json` is a JSON string containing app-specific task parameters. Format of these parameters is entirely up to the application developer.
+  - Will only be called once with given `task_id`.
+  - Can assume `{task_id}/{constants.TASK_INPUTS_DIR}` contains all the resources provided by task creator.
+  - Returns `env_id` and `prerequisites_json` specifying the environment and prerequisites required for providers to compute the task. See environments section for details.
 - `NextSubtask`
-  - takes one argument `task_id`
-  - can assume `CreateTask` was called earlier with the same `task_id`
-  - returns `subtask_id` which has to be a string without whitespaces and slashes (`/`) but the same string cannot be returned more than once
-  - also returns `subtask_params_json` which is the JSON string containing subtask specific parameters
+  - Takes two arguments: `task_id` and `opaque_node_id`.
+  - `opaque_node_id` is an identifier of the node which is going to compute the requested subtask. 'Opaque' means that the identifier doesn't allow to obtain any further information about the node (e.g. public key, IP address).
+  - Can assume `CreateTask` was called earlier with the same `task_id`.
+  - Can return an empty message meaning that the app refuses to assign a subtask to the provider node (for whatever reason).
+  - Returns `subtask_id` which has to be a string without whitespaces and slashes (`/`) but the same string cannot be returned more than once.
+  - Also returns `subtask_params_json` which is the JSON string containing subtask specific parameters.
+  - Also returns `resources` which is a list of names of files required for computing the subtask. Files with these names are required to be present in `{task_id}/{constants.SUBTASK_INPUTS_DIR}` directory.
 - `HasPendingSubtasks`
-  - takes one argument `task_id`
-  - returns a boolean indicating whether there are any more pending subtasks waiting for computation at given moment
-  - in case when it returns `true`, the next `NextSubtask` call should return successfully
+  - Takes one argument `task_id`.
+  - Returns a boolean indicating whether there are any more pending subtasks waiting for computation at given moment.
+  - In case when it returns `true`, the next `NextSubtask` call should return successfully (it can return an empty message altough).
```

Comment:
```suggestion
  - In case when it returns `true`, the next `NextSubtask` call should return successfully (although it can still return an empty message).
```